### PR TITLE
Enhancement/kal datepicker/replace luxon by dayjs

### PR DESCRIPTION
**Pull request by jsp-up**

### CODE REVIEW

**Comment by jsp-up** [comment]
Close #231

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-date.ts [hunk 1]
```diff
@@ -13,160 +20,162 @@ export function coerceKalDateProperty(rawDate: KalDateType): KalDate {
...
+   * Whether the current date is between the `start` a,d `end` dates.
+   * @see https://github.com/iamkun/dayjs/blob/dev/docs/en/Plugin.md#isbetween
    */
   isBetween(start: KalDateType,
-            end: KalDateType) {
-    start = coerceKalDateProperty(start).getDate();
-    end = coerceKalDateProperty(end).getDate();
-    return Interval.fromDateTimes(start, end).contains(this.value);
+            end: KalDateType,
+            exlusions = {start: false, end: false}): boolean {
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-date.ts]: ex**c**lusion , with a `c`

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts [hunk 2]
```diff
@@ -21,15 +21,22 @@ import { DOCUMENT } from '@angular/common';
...
+import { KalInputComponent } from '../kal-input/kal-input.component';
 import { Coerce } from '../../utils/decorators/coerce';
 import { AutoUnsubscribe } from '../../utils/decorators/auto-unsubscribe';
-import { KalInputComponent } from '../kal-input/kal-input.component';
+import { capitalize } from '../../utils/helpers/strings';
+
+/**
+ * Configure DayJS
+ */
+dayjs.extend(localeData);
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts:39]: already present in kal-month-calendar.component.ts

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts [hunk 3]
```diff
@@ -10,10 +10,21 @@ import {
...
 import { KalDatepickerComponent } from '../kal-datepicker.component';
 import { KalDate } from '../kal-date';
+import { DateUnits } from '../kal-datepicker-multi-view/kal-datepicker-multi-view.component';
+import { capitalize } from '../../../utils/helpers/strings';
+import { move } from '../../../utils/helpers/arrays';
+
+/**
+ * Configure DayJS
+ */
+dayjs.extend(localeData);
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts]: already present in kal-datepicker.component.ts

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts [hunk 4]
```diff
@@ -45,23 +50,32 @@ export class KalMonthCalendarComponent implements OnInit {
...
 
     return datesList;
   }
 
+  /**
+   * Returns an array of standalone narrowed weekdays.
+   * @example ['M', 'T', ...]
+   */
+  get narrowWeekDays(): string[] {
+    const days: string[] = dayjs().localeData().weekdaysMin().map(day => capitalize(day).charAt(0));
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts]: day.chartAt(0).toUpperCase() , no need for capitalize

File: projects/kalidea/kaligraphi/src/lib/utils/helpers/arrays.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Moves an array item from one position to another.
+ * Note: This is a pure function so a new array will be returned, instead of altering the array argument.
+ * @see https://github.com/granteagon/move
+ */
+export function move(array: any[], moveIndex: number, toIndex: number) {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/utils/helpers/arrays.ts:6]: function move(array: string[], start: number, end: number) {
    const startElement = array.splice(start, 1);
    array.splice(end, 0, ...startElement);
    return array;
}

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts [hunk 6]
```diff
@@ -88,15 +100,15 @@ export class KalMonthCalendarComponent implements OnInit {
...
+    this.displayedDate = this.displayedDate.add(amount, 'month');
     this.cdr.markForCheck();
   }
 
   /**
    * Change displayed month according to the selected month and the selected year.
    */
-  updateDate(dateUnit: DateObjectUnits) {
-    this.displayedDate = this.displayedDate.set(dateUnit);
+  updateDate(dateUnit: DateUnits) {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-month-calendar/kal-month-calendar.component.ts]: you just use value {value}: DateUnits

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

**Review by ydomenjoud — APPROVED** [review]

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**